### PR TITLE
feat: Idp 1219 sviluppo be - Modifica createTransaction payment

### PR DESCRIPTION
**Pull request by GiovanaSolorzano**

### CODE REVIEW

File: src/main/java/it/gov/pagopa/payment/repository/TransactionInProgressRepositoryExtImpl.java [hunk 1]
```diff
@@ -27,21 +28,29 @@ public class TransactionInProgressRepositoryExtImpl implements TransactionInProg
     private final long trxThrottlingSeconds;
     private final long authorizationExpirationMinutes;
     private final long cancelExpirationMinutes;
+    private final String qrcodePgnBaseUrl;
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payment/repository/TransactionInProgressRepositoryExtImpl.java:31]: there is a typo
```suggestion
    private final String qrcodePngBaseUrl;
```

File: src/main/java/it/gov/pagopa/payment/service/MerchantTransactionServiceImpl.java [hunk 2]
```diff
@@ -61,7 +61,9 @@ public MerchantTransactionsListDTO getMerchantTransactions(String merchantId, St
                                     transaction.getTrxDate().toLocalDateTime(),
                                     authorizationExpirationMinutes,
                                     transaction.getUpdateDate(),
-                                    transaction.getStatus().toString()
+                                    transaction.getStatus().toString(),
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payment/service/MerchantTransactionServiceImpl.java:64]: if inside this application this field is an enum, why inside the DTO returned you are change it into String?

**Review by antonioT90 — CHANGES_REQUESTED** [review]

File: src/main/java/it/gov/pagopa/payment/service/MerchantTransactionServiceImpl.java (hunk 2 above)
Comment (GiovanaSolorzano) [reply src/main/java/it/gov/pagopa/payment/service/MerchantTransactionServiceImpl.java:71]: I didn't modify this field, in the DTO it was already present in the form of String

File: src/main/java/it/gov/pagopa/payment/dto/mapper/TransactionInProgress2TransactionResponseMapper.java [hunk 3]
```diff
@@ -2,19 +2,21 @@
...
+@Slf4j
 public class TransactionInProgress2TransactionResponseMapper
-    implements Function<TransactionInProgress, TransactionResponse> {
+    implements TriFunction<TransactionInProgress, String, String, TransactionResponse> {
 
   @Value("${app.qrCode.expirations.authorizationMinutes}") int authorizationExpirationMinutes;
 
   @Override
-  public TransactionResponse apply(TransactionInProgress transactionInProgress) {
+  public TransactionResponse apply(TransactionInProgress transactionInProgress, String qrcodeImgBaseUrl, String qrcodeTxtBaseUrl) {
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payment/dto/mapper/TransactionInProgress2TransactionResponseMapper.java:19]: let's add inside this Service the injection of qrcodeImgBaseUrl and qrcodeTxtBaseUrl

File: src/main/java/it/gov/pagopa/payment/dto/mapper/TransactionCreationRequest2TransactionInProgressMapper.java [hunk 4]
```diff
@@ -31,21 +33,21 @@ public TransactionInProgress apply(
...
+        .idTrxIssuer(idTrxIssuer)
         .initiativeId(transactionCreationRequest.getInitiativeId())
         .initiativeName(merchantDetail.getInitiativeName())
         .businessName(merchantDetail.getBusinessName())
         .mcc(transactionCreationRequest.getMcc())
         .vat(merchantDetail.getVatNumber())
-        .trxDate(transactionCreationRequest.getTrxDate())
-        .trxChargeDate(transactionCreationRequest.getTrxDate())
+        .trxDate(elaborationTrxDate)
+        .trxChargeDate(elaborationTrxDate)
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payment/dto/mapper/TransactionCreationRequest2TransactionInProgressMapper.java:43]: don't ask the elaborationTrxDate as parameter!
it's a mapper logic to fill it with the timestamp when creating

**Review by antonioT90 — CHANGES_REQUESTED** [review]

**Pull request merged**